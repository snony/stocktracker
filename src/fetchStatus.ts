export default {
  PENDING: 'PENDING',
  SUCCESS: 'SUCCESS',
  FAILED: 'FAILED'
}
