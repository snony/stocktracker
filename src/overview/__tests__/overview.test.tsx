import { mockGlobalState } from '__mock__/globalstate.mock'
import fetchStatus from 'fetchStatus'
import Overview from 'overview/overview'
import React from 'react'
import renderer from 'react-test-renderer'

describe('overview component', () => {
  it('should render correctly before any data fetch completes', () => {
    const mockInitialState = {
      ...mockGlobalState.overview,
      fetchStatus: fetchStatus.INITIAL
    }
    const tree = renderer.create(<Overview overview={mockInitialState} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly on data fetch success', () => {
    const tree = renderer.create(<Overview overview={mockGlobalState.overview} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly on data fetch failed', () => {
    const mockFailedState = {
      ...mockGlobalState.overview,
      fetchStatus: fetchStatus.FAILED
    }
    const tree = renderer.create(<Overview overview={mockFailedState} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
