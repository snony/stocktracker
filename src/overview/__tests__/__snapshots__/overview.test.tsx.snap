// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview component should render correctly 1`] = `
Array [
  .emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 1.4rem;
}

<span
    className="emotion-0"
  >
    Apple Inc.
     (
    AAPL
    )
  </span>,
  .emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 0.9rem;
  font-style: italic;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
    href="http://www.apple.com"
  >
    <span
      className="emotion-0"
    >
      http://www.apple.com
    </span>
  </a>,
  .emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
}

<span
    className="emotion-0"
  >
    Apple Inc is designs, manufactures and markets mobile communication and media devices and personal computers, and sells a variety of related software, services, accessories, networking solutions and third-party digital content and applications.
  </span>,
]
`;
